docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1554 @@
+    Note over Host VRF/Kernel Node A: BGP route lookup
+    Host VRF/Kernel Node A->>Host VRF/Kernel Node A: Match BGP route in host table: 10.128.2.0/24 via 172.18.0.3 (Node B's IP)
+    Host VRF/Kernel Node A->>Underlay Physical Network: Route packet out via breth0
+    Underlay Physical Network->>Node B: Packet arrives at Node B
+    Note over Node B: Traffic is delivered to Pod2
+```
+
+And the reply path:
+
+```mermaid
+sequenceDiagram
+    participant Node B
+    participant Underlay Physical Network
+    participant Host VRF/Kernel Node A
+    participant ovn_cluster_router Node A
+    participant Pod1 on Node A
+    
+    Note over Node B: Traffic is delivered from Pod2
+    Node B->>Underlay Physical Network: Packet leaves Node B
+    Underlay Physical Network->>Host VRF/Kernel Node A: Route packet in via breth0
+    Note over Host VRF/Kernel Node A: route lookup
+    Host VRF/Kernel Node A->>Host VRF/Kernel Node A: Match static route: 10.128.0.0/24 via 100.64.0.1 (Node A's pod subnet)
+    Host VRF/Kernel Node A ->>ovn_cluster_router Node A: forwarded via mp0
+    Note over ovn_cluster_router Node A: route lookup
+    ovn_cluster_router Node A->>ovn_cluster_router Node A: Match the direct route to local pods
+    ovn_cluster_router Node A->>Pod1 on Node A: Packet to local Pod1 (10.128.0.5) via the node switch
+```
+
+#### Import Routes to NodeSubnets
+
+Changes to BGP behavior are necessary to import node subnet routes into the
+gateway router and the host routing table of each node. A new `transport` key
+will be included in the OVN CNI NetConf (the NAD's spec.config JSON) to indicate
+the east-west transport protocol of the network. If a network operates in
+no-overlay mode, this key is set to `noOverlay`. Additionally, the `outboundSNAT`
+key will be included to indicate the SNAT behavior for outbound traffic from pods.
+This information can then be passed into the NetInfo object and utilized by the
+route import and gateway controller of ovnkube-controller.
+
+Note: Users must not set these fields manually; OVN-Kubernetes manages them.
+
+An example OVN-Kubernetes NAD may look like:
+
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: l3-network
+  namespace: default
+spec:
+  config: |
+    {
+            "cniVersion": "0.3.1",
+            "name": "l3-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer3",
+            "mtu": 1500,
+            "netAttachDefName": "default/l3-network",
+            "role": "primary",
+            "transport": "noOverlay",
+            "outboundSNAT": "disabled"
+    }
+```
+
+##### Host Routing Tables
+
+To [maintain isolation between
+UDNs](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5140), node BGP
+speakers do not accept routes to pod subnets by default. However, to route
+east-west traffic in no-overlay mode, these routes must be imported to the host
```

Comment:
super nit: I'd clarify that we would thus import remote node pod routes *if no-overlay is enabled for a given network* and otherwise behaviour is unchanged.